diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3bf3cfdf33..11f422199a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -186,13 +186,13 @@ list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules")
 
 # Don't look for libxml if we're using MSan, since uninstrumented third party
 # code may call MSan interceptors like strlen, leading to false positives.
-if(NOT LLVM_USE_SANITIZER MATCHES "Memory.*")
-  set (LIBXML2_FOUND 0)
-  find_package(LibXml2 2.5.3 QUIET)
-  if (LIBXML2_FOUND)
-    set(CLANG_HAVE_LIBXML 1)
-  endif()
-endif()
+# if(NOT LLVM_USE_SANITIZER MATCHES "Memory.*")
+#   set (LIBXML2_FOUND 0)
+#   find_package(LibXml2 2.5.3 QUIET)
+#   if (LIBXML2_FOUND)
+#     set(CLANG_HAVE_LIBXML 1)
+#   endif()
+# endif()
 
 include(CheckIncludeFile)
 check_include_file(sys/resource.h CLANG_HAVE_RLIMITS)
