--- CMakeLists.txt      2017-11-04 19:49:35.315032166 -0500
+++ CMakeLists.txt      2017-11-04 19:50:00.259020119 -0500
@@ -181,11 +181,6 @@
 # we can include cmake files from this directory.
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules")
 
-find_package(LibXml2 2.5.3 QUIET)
-if (LIBXML2_FOUND)
-  set(CLANG_HAVE_LIBXML 1)
-endif()
-
 find_package(Z3 4.5)
 
 include(CheckIncludeFile)

