From 01756ef52d63bd9762d0b39362e5edd5e56073d8 Mon Sep 17 00:00:00 2001
From: Axel Naumann <Axel.Naumann@cern.ch>
Date: Mon, 3 Mar 2014 23:35:20 +0100
Subject: [PATCH 02/62] Only reset extra parsers if not temporary.

---
 lib/Parse/Parser.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/lib/Parse/Parser.cpp b/lib/Parse/Parser.cpp
index becdc36d60..e4c216ee1f 100644
--- a/lib/Parse/Parser.cpp
+++ b/lib/Parse/Parser.cpp
@@ -422,9 +422,11 @@ Parser::~Parser() {
 
   resetPragmaHandlers();
 
-  PP.removeCommentHandler(CommentSemaHandler.get());
+  if (!IsTemporary)
+    PP.removeCommentHandler(CommentSemaHandler.get());
 
-  PP.clearCodeCompletionHandler();
+  if (!IsTemporary)
+    PP.clearCodeCompletionHandler();
 
   if (getLangOpts().DelayedTemplateParsing &&
       !PP.isIncrementalProcessingEnabled() && !TemplateIds.empty()) {
-- 
2.18.0

