From 528d5aa8d08e234efc78692e6fd372101c53ef33 Mon Sep 17 00:00:00 2001
From: Axel Naumann <Axel.Naumann@cern.ch>
Date: Tue, 7 Jan 2014 12:05:19 +0100
Subject: [PATCH 22/62] Make sure the inode did not get updated since we read
 it.

---
 lib/Basic/FileManager.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/Basic/FileManager.cpp b/lib/Basic/FileManager.cpp
index 4307222802..3fb572be3b 100644
--- a/lib/Basic/FileManager.cpp
+++ b/lib/Basic/FileManager.cpp
@@ -305,6 +305,8 @@ const FileEntry *FileManager::getFile(StringRef Filename, bool openFile,
     // multiple names.
     if (DirInfo != UFE.Dir && Data.IsVFSMapped)
       UFE.Dir = DirInfo;
+  }
+  if (UFE.isValid() && Data.ModTime == UFE.ModTime) {
 
     // Always update the name to use the last name by which a file was accessed.
     // FIXME: Neither this nor always using the first name is correct; we want
-- 
2.18.0

