From f53f612c3496f324bec87be3ae300c4fcd77205e Mon Sep 17 00:00:00 2001
From: Axel Naumann <Axel.Naumann@cern.ch>
Date: Mon, 20 Oct 2014 14:15:26 +0200
Subject: [PATCH 26/62] Accessor to "is file opened"; this is crucial info for
 us.

---
 include/clang/Basic/FileManager.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/clang/Basic/FileManager.h b/include/clang/Basic/FileManager.h
index f6c0b9a40b..6418bef132 100644
--- a/include/clang/Basic/FileManager.h
+++ b/include/clang/Basic/FileManager.h
@@ -84,6 +84,7 @@ public:
 
   StringRef getName() const { return Name; }
   StringRef tryGetRealPathName() const { return RealPathName; }
+  bool isOpen() const { return (bool)File; }
   bool isValid() const { return IsValid; }
   off_t getSize() const { return Size; }
   unsigned getUID() const { return UID; }
-- 
2.18.0

