From 9394eb37b59b80084bc1a7b2558fd3f5c123d495 Mon Sep 17 00:00:00 2001
From: Philippe Canal <pcanal@fnal.gov>
Date: Sat, 17 Oct 2015 01:05:32 -0500
Subject: [PATCH 38/62] Inject DeclUnloader as a friend.

---
 include/clang/AST/DeclCXX.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/clang/AST/DeclCXX.h b/include/clang/AST/DeclCXX.h
index 2f735c5506..a27a171464 100644
--- a/include/clang/AST/DeclCXX.h
+++ b/include/clang/AST/DeclCXX.h
@@ -1842,6 +1842,7 @@ public:
   friend class ASTRecordWriter;
   friend class ASTReader;
   friend class ASTWriter;
+  friend class DeclUnloader;
 };
 
 /// \brief Represents a C++ deduction guide declaration.
-- 
2.18.0

