From 33f7801be5c3b4d731b08ee3ce4346a633165d4a Mon Sep 17 00:00:00 2001
From: Axel Naumann <Axel.Naumann@cern.ch>
Date: Mon, 23 Mar 2015 15:22:49 +0100
Subject: [PATCH 39/62] Do not load the source file just to get an irrelevant
 SourceLoc (ROOT-7111).

---
 lib/Lex/Lexer.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lib/Lex/Lexer.cpp b/lib/Lex/Lexer.cpp
index 61bcef8cb7..dbe6b4db94 100644
--- a/lib/Lex/Lexer.cpp
+++ b/lib/Lex/Lexer.cpp
@@ -770,6 +770,10 @@ SourceLocation Lexer::getLocForEndOfToken(SourceLocation Loc, unsigned Offset,
       return SourceLocation(); // Points inside the macro expansion.
   }
 
+  // Don't hit the file system for ASTReader tokens.
+  if (SM.isLoadedSourceLocation(Loc))
+    return Loc;
+
   unsigned Len = Lexer::MeasureTokenLength(Loc, SM, LangOpts);
   if (Len > Offset)
     Len = Len - Offset;
-- 
2.18.0

