From 44fc39f5132915a7dc65365c8a815d948c6cec05 Mon Sep 17 00:00:00 2001
From: Axel Naumann <Axel.Naumann@cern.ch>
Date: Fri, 22 Apr 2016 15:59:00 +0200
Subject: [PATCH 40/62] Construct DestroyTemplateIdAnnotationsRAIIObj given a
 Parser.

---
 include/clang/Parse/RAIIObjectsForParser.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/clang/Parse/RAIIObjectsForParser.h b/include/clang/Parse/RAIIObjectsForParser.h
index 0422b038da..58a23074d8 100644
--- a/include/clang/Parse/RAIIObjectsForParser.h
+++ b/include/clang/Parse/RAIIObjectsForParser.h
@@ -453,6 +453,9 @@ namespace clang {
         SmallVectorImpl<TemplateIdAnnotation *> &Container)
         : Container(Container) {}
 
+    DestroyTemplateIdAnnotationsRAIIObj(Parser& P)
+        : Container(P.TemplateIds) {}
+
     ~DestroyTemplateIdAnnotationsRAIIObj() {
       for (SmallVectorImpl<TemplateIdAnnotation *>::iterator I =
                Container.begin(),
-- 
2.18.0

