From be1d7e867253a6223f2c89b85c70dd281d6f5ddb Mon Sep 17 00:00:00 2001
From: Axel Naumann <Axel.Naumann@cern.ch>
Date: Mon, 7 Nov 2016 15:03:38 +0100
Subject: [PATCH 48/62] In the RAII ctor, set private Parser fields to default
 values.

---
 lib/Sema/SemaCXXScopeSpec.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lib/Sema/SemaCXXScopeSpec.cpp b/lib/Sema/SemaCXXScopeSpec.cpp
index 6da4d2a261..12d0786d74 100644
--- a/lib/Sema/SemaCXXScopeSpec.cpp
+++ b/lib/Sema/SemaCXXScopeSpec.cpp
@@ -108,6 +108,10 @@ DeclContext *Sema::computeDeclContext(const CXXScopeSpec &SS,
           // into that class template definition.
           QualType Injected
             = ClassTemplate->getInjectedClassNameSpecialization();
+
+          // Injected might not be canonical
+          Injected = Injected.getCanonicalType();
+
           if (Context.hasSameType(Injected, ContextType))
             return ClassTemplate->getTemplatedDecl();
 
-- 
2.18.0

