From 2df1810da235bc09866dd14afdd0c2d5b78104f0 Mon Sep 17 00:00:00 2001
From: Axel Naumann <Axel.Naumann@cern.ch>
Date: Fri, 31 Mar 2017 20:54:44 +0200
Subject: [PATCH 56/62] Clear also EmittedDeferredDecls.

---
 lib/CodeGen/CodeGenModule.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/CodeGen/CodeGenModule.cpp b/lib/CodeGen/CodeGenModule.cpp
index c41fe622df..18f9d1d14a 100644
--- a/lib/CodeGen/CodeGenModule.cpp
+++ b/lib/CodeGen/CodeGenModule.cpp
@@ -359,6 +359,7 @@ void CodeGenModule::checkAliases() {
 
 void CodeGenModule::clear() {
   DeferredDeclsToEmit.clear();
+  EmittedDeferredDecls.clear();
   if (OpenMPRuntime)
     OpenMPRuntime->clear();
 }
-- 
2.18.0

