From b1c04fd243257bb2622a522ed566795516096a1e Mon Sep 17 00:00:00 2001
From: Axel Naumann <Axel.Naumann@cern.ch>
Date: Mon, 3 Mar 2014 23:35:20 +0100
Subject: [PATCH 02/83] Only reset extra parsers if not temporary.

---
 lib/Parse/Parser.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/lib/Parse/Parser.cpp b/lib/Parse/Parser.cpp
index 6481992ffe..ed9aae4906 100644
--- a/lib/Parse/Parser.cpp
+++ b/lib/Parse/Parser.cpp
@@ -469,9 +469,11 @@ Parser::~Parser() {
 
   resetPragmaHandlers();
 
-  PP.removeCommentHandler(CommentSemaHandler.get());
+  if (!IsTemporary)
+    PP.removeCommentHandler(CommentSemaHandler.get());
 
-  PP.clearCodeCompletionHandler();
+  if (!IsTemporary)
+    PP.clearCodeCompletionHandler();
 
   DestroyTemplateIds();
 }
-- 
2.34.1

