From 8f6d8113077164c3ca1ce76d892ae8f6f28fc753 Mon Sep 17 00:00:00 2001
From: Axel Naumann <Axel.Naumann@cern.ch>
Date: Mon, 20 Oct 2014 14:15:26 +0200
Subject: [PATCH 21/83] Accessor to "is file opened"; this is crucial info for
 us.

---
 include/clang/Basic/FileEntry.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/clang/Basic/FileEntry.h b/include/clang/Basic/FileEntry.h
index 6e91b42e18..8b745fc976 100644
--- a/include/clang/Basic/FileEntry.h
+++ b/include/clang/Basic/FileEntry.h
@@ -365,6 +365,7 @@ public:
   FileEntryRef getLastRef() const { return *LastRef; }
 
   StringRef tryGetRealPathName() const { return RealPathName; }
+  bool isOpen() const { return (bool)File; }
   bool isValid() const { return IsValid; }
   off_t getSize() const { return Size; }
   unsigned getUID() const { return UID; }
-- 
2.34.1

