From f19f0f96301a36761cfd95d6cbb1d38bd0e1cafa Mon Sep 17 00:00:00 2001
From: Axel Naumann <Axel.Naumann@cern.ch>
Date: Thu, 28 May 2015 11:43:50 +0200
Subject: [PATCH 26/73] Clear pending decls from Builder in case of errors
 (ROOT-7295).

---
 clang/lib/CodeGen/ModuleBuilder.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/clang/lib/CodeGen/ModuleBuilder.cpp b/clang/lib/CodeGen/ModuleBuilder.cpp
index aeccf768f7ef..3ea89fb17670 100644
--- a/clang/lib/CodeGen/ModuleBuilder.cpp
+++ b/clang/lib/CodeGen/ModuleBuilder.cpp
@@ -104,6 +104,11 @@ namespace clang {
     }
 
     llvm::Module *ReleaseModule() {
+      // Remove pending etc decls in case of error; the asserts in StartModule()
+      // will rightfully be confused otherwise, as none of the decls were
+      // emitted.
+      if (Diags.hasErrorOccurred())
+        Builder->clear();
       return M.release();
     }
 
-- 
2.42.0

