From 5f3b687edb87ce59476c0a47d683bf9fc2c3cb30 Mon Sep 17 00:00:00 2001
From: Axel Naumann <Axel.Naumann@cern.ch>
Date: Mon, 20 Oct 2014 14:15:26 +0200
Subject: [PATCH 29/66] Accessor to "is file opened"; this is crucial info for
 us.

---
 clang/include/clang/Basic/FileEntry.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/clang/include/clang/Basic/FileEntry.h b/clang/include/clang/Basic/FileEntry.h
index 93e84cf233a3..7af5b2bb937e 100644
--- a/clang/include/clang/Basic/FileEntry.h
+++ b/clang/include/clang/Basic/FileEntry.h
@@ -388,6 +388,7 @@ public:
   FileEntryRef getLastRef() const { return *LastRef; }
 
   StringRef tryGetRealPathName() const { return RealPathName; }
+  bool isOpen() const { return (bool)File; }
   off_t getSize() const { return Size; }
   unsigned getUID() const { return UID; }
   const llvm::sys::fs::UniqueID &getUniqueID() const { return UniqueID; }
