From efb9ca3d04327c9d4cf5e44dbe8df1aa41aaa215 Mon Sep 17 00:00:00 2001
From: Axel Naumann <Axel.Naumann@cern.ch>
Date: Mon, 23 Mar 2015 15:22:49 +0100
Subject: [PATCH 31/83] Do not load the source file just to get an irrelevant
 SourceLoc (ROOT-7111).

---
 lib/Lex/Lexer.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lib/Lex/Lexer.cpp b/lib/Lex/Lexer.cpp
index 64944492eb..94d9f1a5c4 100644
--- a/lib/Lex/Lexer.cpp
+++ b/lib/Lex/Lexer.cpp
@@ -791,6 +791,10 @@ SourceLocation Lexer::getLocForEndOfToken(SourceLocation Loc, unsigned Offset,
       return {}; // Points inside the macro expansion.
   }
 
+  // Don't hit the file system for ASTReader tokens.
+  if (SM.isLoadedSourceLocation(Loc))
+    return Loc;
+
   unsigned Len = Lexer::MeasureTokenLength(Loc, SM, LangOpts);
   if (Len > Offset)
     Len = Len - Offset;
-- 
2.34.1

