From 7af8c82133913a1d46ecc2be337fde4aba2c4f7e Mon Sep 17 00:00:00 2001
From: Axel Naumann <Axel.Naumann@cern.ch>
Date: Thu, 28 May 2015 11:43:50 +0200
Subject: [PATCH 32/65] Clear pending decls from Builder in case of errors
 (ROOT-7295).

---
 clang/lib/CodeGen/ModuleBuilder.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/clang/lib/CodeGen/ModuleBuilder.cpp b/clang/lib/CodeGen/ModuleBuilder.cpp
index 8b9bbc960a51..0f2ded1301af 100644
--- a/clang/lib/CodeGen/ModuleBuilder.cpp
+++ b/clang/lib/CodeGen/ModuleBuilder.cpp
@@ -107,6 +107,11 @@ namespace clang {
     }
 
     llvm::Module *ReleaseModule() {
+      // Remove pending etc decls in case of error; the asserts in StartModule()
+      // will rightfully be confused otherwise, as none of the decls were
+      // emitted.
+      if (Diags.hasErrorOccurred())
+        Builder->clear();
       return M.release();
     }
 
