From 0eef4f512833eb001542e438edaba0b1a93d949c Mon Sep 17 00:00:00 2001
From: Axel Naumann <Axel.Naumann@cern.ch>
Date: Mon, 7 Nov 2016 15:03:38 +0100
Subject: [PATCH 35/83] In the RAII ctor, set private Parser fields to default
 values.

---
 lib/Sema/SemaCXXScopeSpec.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lib/Sema/SemaCXXScopeSpec.cpp b/lib/Sema/SemaCXXScopeSpec.cpp
index 1c8f6329bd..5dfe5c0f29 100644
--- a/lib/Sema/SemaCXXScopeSpec.cpp
+++ b/lib/Sema/SemaCXXScopeSpec.cpp
@@ -107,6 +107,10 @@ DeclContext *Sema::computeDeclContext(const CXXScopeSpec &SS,
           // into that class template definition.
           QualType Injected
             = ClassTemplate->getInjectedClassNameSpecialization();
+
+          // Injected might not be canonical
+          Injected = Injected.getCanonicalType();
+
           if (Context.hasSameType(Injected, ContextType))
             return ClassTemplate->getTemplatedDecl();
 
-- 
2.34.1

