From 71f9d58176711375c2be356cfdb2103b99581382 Mon Sep 17 00:00:00 2001
From: Axel Naumann <Axel.Naumann@cern.ch>
Date: Thu, 18 Sep 2014 23:40:14 +0200
Subject: [PATCH 58/83] Silence enum typed/untyped redecl until fwd decl can
 handle it.

---
 lib/Sema/SemaDecl.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/Sema/SemaDecl.cpp b/lib/Sema/SemaDecl.cpp
index d7891caa82..1ae6e9eacc 100644
--- a/lib/Sema/SemaDecl.cpp
+++ b/lib/Sema/SemaDecl.cpp
@@ -15284,9 +15284,11 @@ bool Sema::CheckEnumRedeclaration(SourceLocation EnumLoc, bool IsScoped,
       return true;
     }
   } else if (IsFixed != Prev->isFixed()) {
+#if 0
     Diag(EnumLoc, diag::err_enum_redeclare_fixed_mismatch)
       << Prev->isFixed();
     Diag(Prev->getLocation(), diag::note_previous_declaration);
+#endif
     return true;
   }
 
-- 
2.34.1

