From ebc35f652fe1f71979ccaa16e1813ce1d500cdeb Mon Sep 17 00:00:00 2001
From: Axel Naumann <Axel.Naumann@cern.ch>
Date: Mon, 6 Oct 2014 10:36:27 +0200
Subject: [PATCH 61/83] Do not read the file to determine its name!

---
 lib/Basic/SourceManager.cpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/lib/Basic/SourceManager.cpp b/lib/Basic/SourceManager.cpp
index 8c17fb8090..42abc2963c 100644
--- a/lib/Basic/SourceManager.cpp
+++ b/lib/Basic/SourceManager.cpp
@@ -1524,6 +1524,15 @@ StringRef SourceManager::getBufferName(SourceLocation Loc,
   auto B = getBufferOrNone(getFileID(Loc));
   if (Invalid)
     *Invalid = !B;
+
+  // Try to get the name without reading the buffer.
+  FileID FID = getFileID(Loc);
+  const SrcMgr::SLocEntry &Entry = getSLocEntry(FID, Invalid);
+  if (!Invalid && Entry.isFile()) {
+    if (const FileEntry* FE = Entry.getFile().getContentCache().ContentsEntry)
+      return FE->getName();
+  }
+
   return B ? B->getBufferIdentifier() : "<invalid buffer>";
 }
 
-- 
2.34.1

