From cdb561190d0962700b1cba96d1ba68886793d594 Mon Sep 17 00:00:00 2001
From: Axel Naumann <Axel.Naumann@cern.ch>
Date: Thu, 4 Feb 2021 11:52:26 +0100
Subject: [PATCH 76/83] When unloading, also forget that the decl was emitted.

---
 lib/CodeGen/ModuleBuilder.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/CodeGen/ModuleBuilder.cpp b/lib/CodeGen/ModuleBuilder.cpp
index 8694bda2df..8d5db56a1b 100644
--- a/lib/CodeGen/ModuleBuilder.cpp
+++ b/lib/CodeGen/ModuleBuilder.cpp
@@ -312,6 +312,7 @@ namespace clang {
 
     void forgetDecl(const GlobalDecl& GD, llvm::StringRef MangledName) {
       Builder->DeferredDecls.erase(MangledName);
+      Builder->Manglings.erase(MangledName);
     }
 
     void Initialize(ASTContext &Context) override {
-- 
2.34.1

