From 6dff23924304037da54f1dc0ee477675f18373cf Mon Sep 17 00:00:00 2001
From: Vassil Vassilev <v.g.vassilev@gmail.com>
Date: Sat, 15 Aug 2020 06:39:01 +0000
Subject: [PATCH 78/83] Add missing include to fix -Dcxxmodules=On

---
 include/clang/Serialization/GlobalModuleIndex.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/clang/Serialization/GlobalModuleIndex.h b/include/clang/Serialization/GlobalModuleIndex.h
index 5f48126262..859c79ddb3 100644
--- a/include/clang/Serialization/GlobalModuleIndex.h
+++ b/include/clang/Serialization/GlobalModuleIndex.h
@@ -16,6 +16,7 @@
 #define LLVM_CLANG_SERIALIZATION_GLOBALMODULEINDEX_H
 
 #include "llvm/ADT/DenseMap.h"
+#include <llvm/ADT/DenseSet.h>
 #include "llvm/ADT/SmallPtrSet.h"
 #include "llvm/ADT/SmallVector.h"
 #include "llvm/ADT/StringMap.h"
-- 
2.34.1

