From 27a4788ec47b6a271f539265368c45b4b4bae3a9 Mon Sep 17 00:00:00 2001
From: Axel Naumann <Axel.Naumann@cern.ch>
Date: Tue, 27 Sep 2016 11:55:47 +0200
Subject: [PATCH 089/107] Improve int literal printing for template args. To be
 upstreamed.

---
 lib/AST/TemplateBase.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/lib/AST/TemplateBase.cpp b/lib/AST/TemplateBase.cpp
index cb4cbd2f76..d6339acdf8 100644
--- a/lib/AST/TemplateBase.cpp
+++ b/lib/AST/TemplateBase.cpp
@@ -76,6 +76,12 @@ static void printIntegral(const TemplateArgument &TemplArg,
     Out << "'";
   } else {
     Out << Val;
+    // Handle cases where the value is too large to fit into the underlying type
+    // i.e. where the unsignedness matters.
+    if (T->isBuiltinType()) {
+      if (Val.isUnsigned() && Val.getBitWidth() == 64 && Val.isNegative())
+        Out << "ull";
+    }
   }
 }
 
-- 
2.35.1

