From 9b00ebdc8a19663ef380f3dc5af3ad643c8abdae Mon Sep 17 00:00:00 2001
From: Axel Naumann <Axel.Naumann@cern.ch>
Date: Tue, 6 Sep 2016 15:30:23 +0200
Subject: [PATCH 090/110] Spaces.

---
 lib/CodeGen/ModuleBuilder.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/CodeGen/ModuleBuilder.cpp b/lib/CodeGen/ModuleBuilder.cpp
index 748ed13fff..d2c6d8b0b2 100644
--- a/lib/CodeGen/ModuleBuilder.cpp
+++ b/lib/CodeGen/ModuleBuilder.cpp
@@ -294,7 +294,7 @@ namespace clang {
     }
 
     void forgetGlobal(llvm::GlobalValue* GV) {
-      for(auto I = Builder->ConstantStringMap.begin(),
+      for (auto I = Builder->ConstantStringMap.begin(),
             E = Builder->ConstantStringMap.end(); I != E; ++I) {
         if (I->second == GV) {
           Builder->ConstantStringMap.erase(I);
-- 
2.29.2

