From 5b69bd706085271321e4125f616927f2176da608 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Tue, 11 May 2021 22:12:48 +0200
Subject: [PATCH] Make clangd CompletionModel not depend on directory layout

---
 clangd/quality/CompletionModel.cmake | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/clangd/quality/CompletionModel.cmake b/clangd/quality/CompletionModel.cmake
index 60c6d2a..dc0c0cd 100644
--- a/clangd/quality/CompletionModel.cmake
+++ b/clangd/quality/CompletionModel.cmake
@@ -4,9 +4,10 @@
 # ${CMAKE_CURRENT_BINARY_DIR}. The generated header
 # will define a C++ class called ${cpp_class} - which may be a
 # namespace-qualified class name.
+set(CLANGD_COMPLETION_MODEL_COMPILER ${CMAKE_CURRENT_LIST_DIR}/CompletionModelCodegen.py)
 function(gen_decision_forest model filename cpp_class)
-  set(model_compiler ${CMAKE_SOURCE_DIR}/../clang-tools-extra/clangd/quality/CompletionModelCodegen.py)
-  
+  set(model_compiler ${CLANGD_COMPLETION_MODEL_COMPILER})
+
   set(output_dir ${CMAKE_CURRENT_BINARY_DIR})
   set(header_file ${output_dir}/${filename}.h)
   set(cpp_file ${output_dir}/${filename}.cpp)
-- 
2.30.2

