From 5e9e968bc3e2d7c666d5ac49c46de0c2593680e3 Mon Sep 17 00:00:00 2001
From: Isuru Fernando <isuruf@gmail.com>
Date: Mon, 8 Apr 2019 16:32:17 -0500
Subject: [PATCH 2/7] Fix sysroot detection for linux

---
 clang/lib/Driver/ToolChains/Linux.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/clang/lib/Driver/ToolChains/Linux.cpp b/clang/lib/Driver/ToolChains/Linux.cpp
index ceb1a982c3a4..d6f62ec0acd6 100644
--- a/clang/lib/Driver/ToolChains/Linux.cpp
+++ b/clang/lib/Driver/ToolChains/Linux.cpp
@@ -394,6 +394,14 @@ std::string Linux::computeSysRoot() const {
   if (getVFS().exists(Path))
     return Path;
 
+  Path =
+      (InstallDir + "/../../../../" + TripleStr + "/sysroot")
+          .str();
+
+  if (getVFS().exists(Path))
+    return Path;
+
+
   Path = (InstallDir + "/../../../../sysroot" + Multilib.osSuffix()).str();
 
   if (getVFS().exists(Path))
-- 
2.37.0.windows.1

