From 907d87a241ef274c25ed964451c2a5b3fa2a644b Mon Sep 17 00:00:00 2001
From: Isuru Fernando <isuruf@gmail.com>
Date: Sat, 27 Jul 2019 11:55:23 -0500
Subject: [PATCH 05/10] Set VERSION in osx as well

---
 clang/tools/libclang/CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/clang/tools/libclang/CMakeLists.txt b/clang/tools/libclang/CMakeLists.txt
index ac7a9a8db37c..c6ffe1ff6775 100644
--- a/clang/tools/libclang/CMakeLists.txt
+++ b/clang/tools/libclang/CMakeLists.txt
@@ -172,6 +172,11 @@ if(ENABLE_SHARED)
 
     set_property(TARGET libclang APPEND_STRING PROPERTY
                  LINK_FLAGS ${LIBCLANG_LINK_FLAGS})
+    set_target_properties(libclang
+      PROPERTIES
+      VERSION ${LIBCLANG_LIBRARY_VERSION}
+      SOVERSION ${LIBCLANG_LIBRARY_VERSION}
+    )
   else()
     set_target_properties(libclang
       PROPERTIES
