From 241bc3171ce4fe95db6339b6c5074d23541d32e0 Mon Sep 17 00:00:00 2001
From: Vassil Vassilev <vvasilev@cern.ch>
Date: Fri, 28 Mar 2014 10:48:12 +0100
Subject: [PATCH 14/61] Access sema internals and revert the list of unused
 decls.

---
 include/clang/Sema/Sema.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/clang/Sema/Sema.h b/include/clang/Sema/Sema.h
index e6c63fd9c0..6080a6a373 100644
--- a/include/clang/Sema/Sema.h
+++ b/include/clang/Sema/Sema.h
@@ -11110,6 +11110,7 @@ protected:
   friend class ASTReader;
   friend class ASTDeclReader;
   friend class ASTWriter;
+  friend class DeclUnloader;
 
 public:
   /// Retrieve the keyword associated
-- 
2.30.1

