From 1b29c04cf5651f3b89bef3027fea4fe74c36a7a8 Mon Sep 17 00:00:00 2001
From: Axel Naumann <Axel.Naumann@cern.ch>
Date: Mon, 23 Mar 2015 15:22:49 +0100
Subject: [PATCH 36/61] Do not load the source file just to get an irrelevant
 SourceLoc (ROOT-7111).

---
 lib/Lex/Lexer.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lib/Lex/Lexer.cpp b/lib/Lex/Lexer.cpp
index db53e6bec0..c8ed42d266 100644
--- a/lib/Lex/Lexer.cpp
+++ b/lib/Lex/Lexer.cpp
@@ -777,6 +777,10 @@ SourceLocation Lexer::getLocForEndOfToken(SourceLocation Loc, unsigned Offset,
       return {}; // Points inside the macro expansion.
   }
 
+  // Don't hit the file system for ASTReader tokens.
+  if (SM.isLoadedSourceLocation(Loc))
+    return Loc;
+
   unsigned Len = Lexer::MeasureTokenLength(Loc, SM, LangOpts);
   if (Len > Offset)
     Len = Len - Offset;
-- 
2.30.1

